plugins/module_utils/crypto/module_backends/csr.py

```diff
@@ -626,8 +626,10 @@ def _check_basicConstraints(extensions):
                 return False
             # Check criticality
             if self.basicConstraints:
-                if bc_ext.critical != self.basicConstraints_critical:
+                if not bc_ext or bc_ext.critical != self.basicConstraints_critical:
```

Comment:
How about inverting the logic and doing:
```suggestion
                return bc_ext is not None and bc_ext.critical == self.basicConstraints_critical
```
and removing the `return False` / `return True` below? I think that makes the code easier to read and the conditions easier to understand.